sherpa/astro/data.py

```diff
@@ -1523,23 +1523,43 @@ def _get_rate(self):
     def _set_rate(self, val):
         self._rate = bool_cast(val)
         for id in self.background_ids:
-            # TODO: shouldn't this store bool_cast(val) instead?
             self.get_background(id).rate = val
 
     rate = property(_get_rate, _set_rate,
-                    doc='Quantity of y-axis: counts or counts/sec')
+                    doc="""Is the Y axis displayed as a rate when plotting data?
+
+When True the y axis is normalised by the exposure time to display
+a rate.""")
 
     def _get_plot_fac(self):
         return self._plot_fac
 
     def _set_plot_fac(self, val):
-        self._plot_fac = int(val)
+        # I'd prefer to check whether val is an integer, but there
+        # may be users who have set the value to 2.0 and it doesn't
+        # seem worth breaking that code.
+        #
+        try:
+            okay = val == int(val)
+        except (ValueError, TypeError):
+            # For when int() can't convert val, which can raise
+            # different errors.
+            okay = False
+
+        if not okay:
+            raise DataErr("bad", "plot_fac setting", val)
+
+        val = int(val)
```

Comment:
```suggestion
        try:
            val == int(val)
        except (ValueError, TypeError):
            # For when int() can't convert val, which can raise
            # different errors.
            raise DataErr("bad", "plot_fac setting", val)
```
I don't understand that extra "okay" variable. If `val = int(val)` works, then val is an int and we don't need okey. If it dies not, then we raise an Error anyway.

Reply:
So, this bit of code went through several revisions, so maybe I just got confused. What is was worried about for your version was that the val that would get used in the `raise` call to create the error message would have already been converted to an integer, and so the message would have been confusing to the user. But I now realize that that thinking is baloney, so I'll make the change.

Reply:
Actually, no. I think I do need the existing code. I want to allow

    factor=1.0

(since this was allowed before, so I don't see it worth breaking code) but disallow

    factor = 1.5

(since this leads to potentially-surprising results) and so you need the current version - or perhaps better would be

```
    try:
        ival = int(val)
        okay = val == ival
    except (...):
        okay = False

    if okay:
        raise ...
```

An alternative to error-ing out would be to create a warning message, but my guess is that not many people have used non-integer-like values (and  that this is technically different to allowing 1.0).